src/app/server/Server.cpp

```diff
@@ -101,6 +106,103 @@ static ::chip::PersistedCounter<chip::EventNumber> sGlobalEventIdCounter;
 static ::chip::app::CircularEventBuffer sLoggingBuffer[CHIP_NUM_EVENT_LOGGING_BUFFERS];
 #endif // CHIP_CONFIG_ENABLE_SERVER_IM_EVENT
 
+#if defined CHIP_CONFIG_TC_REQUIRED_ACKNOWLEDGEMENTS && defined CHIP_CONFIG_TC_REQUIRED_ACKNOWLEDGEMENTS_VERSION
+app::DefaultEnhancedSetupFlowProvider sDefaultEnhancedSetupFlowProviderInstance;
```

Comment:
This is going to add the DefaultEnhancedSetupFlowProvider to all applications that use Server.  Why do we need to support this in CommonCaseDeviceServerInitParams at all?  Things that actually have terms/conditions (like all other non-example SDK consumers) should be using a non-common-case server init params to start with.

Reply:
@swan-amazon Please do not resolve comments without addressing. I have the same question (just below).